app/test_main.py

```diff
@@ -2,0 +1,20 @@
+from typing import Any
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_has_internet_connection_called(
+        mocked_has_internet_connection: Any
+) -> None:
+    can_access_google_page("https://google.com")
+    mocked_has_internet_connection.assert_called_once()
+
+
+@mock.patch("app.main.valid_google_url")
+def test_valid_google_url_called_with_correct_parameters(
+        mocked_valid_google_url: Any
+) -> None:
```

Comment:
The test name `test_valid_google_url_called_with_correct_parameters` could be more descriptive. Consider renaming it to something like `test_can_access_google_page_calls_valid_google_url_with_correct_parameters` to better describe what the test is verifying.